cluster/state.go

```diff
@@ -249,3 +134,106 @@ func GetK8sVersion(localConfigPath string, k8sWrapTransport k8s.WrapTransport) (
 	}
 	return fmt.Sprintf("%#v", *serverVersion), nil
 }
+
+func RebuildState(ctx context.Context, rkeConfig *v3.RancherKubernetesEngineConfig, oldState *RKEFullState, configPath, configDir string) (*RKEFullState, error) {
+	newState := &RKEFullState{
+		DesiredState: RKEState{
+			RancherKubernetesEngineConfig: rkeConfig.DeepCopy(),
+		},
+	}
+
+	// Rebuilding the certificates of the desired state
+	if oldState.DesiredState.CertificatesBundle == nil {
+		// Get the certificate Bundle
+		certBundle, err := pki.GenerateRKECerts(ctx, *rkeConfig, "", "")
+		if err != nil {
+			return nil, fmt.Errorf("Failed to generate certificate bundle: %v", err)
+		}
+		newState.DesiredState.CertificatesBundle = certBundle
+	} else {
+		// Regenerating etcd certificates for any new etcd nodes
+		pkiCertBundle := oldState.DesiredState.CertificatesBundle
+		if err := pki.GenerateEtcdCertificates(ctx, pkiCertBundle, *rkeConfig, "", ""); err != nil {
+			return nil, err
+		}
+		// Regenerating kubeapi certificates for any new kubeapi nodes
+		if err := pki.GenerateKubeAPICertificate(ctx, pkiCertBundle, *rkeConfig, "", ""); err != nil {
+			return nil, err
+		}
+		// Regenerating kubeadmin certificates/config
+		if err := pki.GenerateKubeAdminCertificate(ctx, pkiCertBundle, *rkeConfig, configPath, configDir); err != nil {
+			return nil, err
+		}
+		newState.DesiredState.CertificatesBundle = pkiCertBundle
+	}
+	newState.CurrentState = oldState.CurrentState
+	return newState, nil
+}
+
+func (s *RKEFullState) WriteStateFile(ctx context.Context, statePath string) error {
```

Comment:
lets make all the methods that are called within the package only, start with lowercase letter.

Reply:
we do call this method from cmd package